packages/contracts/contracts/optimistic-ethereum/OVM/predeploys/OVM_SequencerEntrypoint.sol

```diff
@@ -6,137 +6,56 @@ pragma solidity >0.5.0 <0.8.0;
-
-        // Remainder is the transaction to execute.
-        bytes memory compressedTx = Lib_BytesUtils.slice(msg.data, 66);
-        bool isEthSignedMessage = transactionType == TransactionType.ETH_SIGNED_MESSAGE;
-
-        // Grab the chain ID for the current network.
-        uint256 chainId;
-        assembly {
-            chainId := chainid()
-        }
-
-        // Need to decompress and then re-encode the transaction based on the original encoding.
-        bytes memory encodedTx = Lib_OVMCodec.encodeEIP155Transaction(
-            Lib_OVMCodec.decompressEIP155Transaction(
-                compressedTx,
-                chainId
-            ),
-            isEthSignedMessage
+        Lib_EIP155Tx.EIP155Tx memory transaction = Lib_EIP155Tx.decode(
+            msg.data,
+            Lib_ExecutionManagerWrapper.ovmCHAINID()
         );
 
-        address target = Lib_ECDSAUtils.recover(
-            encodedTx,
-            isEthSignedMessage,
-            v,
-            r,
-            s
-        );
+        // Value is computed on the fly. Store it to save some gas.
+        address target = transaction.sender();
 
         bool isEmptyContract;
         assembly {
             isEmptyContract := iszero(extcodesize(target))
         }
 
         if (isEmptyContract) {
-            // ProxyEOA has not yet been deployed for this EOA.
-            bytes32 messageHash = Lib_ECDSAUtils.getMessageHash(encodedTx, isEthSignedMessage);
-            Lib_ExecutionManagerWrapper.ovmCREATEEOA(messageHash, v, r, s);
-        }
-
-        Lib_OVMCodec.EOASignatureType sigtype;
-        if (isEthSignedMessage) {
-            sigtype = Lib_OVMCodec.EOASignatureType.ETH_SIGNED_MESSAGE;
-        } else {
-            sigtype = Lib_OVMCodec.EOASignatureType.EIP155_TRANSACTION;
+            Lib_ExecutionManagerWrapper.ovmCREATEEOA(
+                transaction.hash(),
+                transaction.recoveryParam,
+                transaction.r,
+                transaction.s
+            );
         }
 
-        iOVM_ECDSAContractAccount(target).execute(
-            encodedTx,
-            sigtype,
-            v,
-            r,
-            s
-        );
-    }
-
-
-    /**********************
-     * Internal Functions *
-     **********************/
-
-    /**
-     * Converts a uint256 into a TransactionType enum.
-     * @param _transactionType Transaction type index.
-     * @return _txType Transaction type enum value.
-     */
-    function _getTransactionType(
-        uint8 _transactionType
-    )
-        internal
-        returns (
-            TransactionType _txType
-        )
-    {
-        if (_transactionType == 0) {
-            return TransactionType.NATIVE_ETH_TRANSACTION;
-        } if (_transactionType == 2) {
-            return TransactionType.ETH_SIGNED_MESSAGE;
-        } else {
-            revert("Transaction type must be 0 or 2");
-        }
+        iOVM_ECDSAContractAccount(target).execute(msg.data);
```

Comment:
Since `msg.data` is used up above on line 38, we might get some gas savings if we cache into a `bytes memory` object first so that we are not `CALLDATACOPY`ing to new memory twice.

Reply:
Interesting, I hadn't thought of that. Do we know that `msg.data` performs a `CALLDATACOPY`?

Reply:
Yeah, I think it fundamentally has to.  I suppose it could be smart enough to do it once and hold on if it sees duplicates, but I think that's unlikely.

Reply:
Just tested this, it does indeed perform two `CALLDATACOPY` operations and it's slightly more efficient to store the value. Just pushed a commit that does this.